src/main/java/com/datasonnet/plugins/DefaultYamlFormatPlugin.java

```diff
@@ -0,0 +1,145 @@
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLParser;
+import ujson.Value;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.List;
+
+public class DefaultYamlFormatPlugin extends BaseJacksonDataFormatPlugin {
+
+    public static final String DS_PARAM_YAML_HEADER = "removehead";
+    // this may break some things like reading 3.0.0 as a number,
+    // would need to specify this in docs
+    public static final String DS_PARAM_DISABLE_QUOTES = "disablequotes";
+
+    public DefaultYamlFormatPlugin(){
+        supportedTypes.add(MediaTypes.APPLICATION_YAML);
+
+        readerSupportedClasses.add(java.lang.String.class);
+        readerSupportedClasses.add(java.lang.CharSequence.class);
+        readerSupportedClasses.add(java.nio.ByteBuffer.class);
+        readerSupportedClasses.add(byte[].class);
+
+        writerSupportedClasses.add(java.lang.String.class);
+        writerSupportedClasses.add(java.lang.CharSequence.class);
+        writerSupportedClasses.add(java.nio.ByteBuffer.class);
+        writerSupportedClasses.add(byte[].class);
+
+        readerParams.add(DS_PARAM_YAML_HEADER);
+        writerParams.addAll(readerParams);
+        writerParams.add(DS_PARAM_DISABLE_QUOTES);
+
+    }
+
+    @Override
+    public Value read(Document<?> doc) throws PluginException {
+        if (doc.getContent() == null) {
+            return ujson.Null$.MODULE$;
+        }
+
+        try {
+
+            YAMLFactory yaml = new YAMLFactory();
+            ObjectMapper mapper = new ObjectMapper();
+
+            YAMLParser yamlParser = yaml.createParser((String) doc.getContent());
+            List<JsonNode> docs = mapper.readValues(yamlParser, new TypeReference<JsonNode>() {}).readAll();
+
+            if(docs.size()<=1){ //if only one node, only one object so dont return the list
+                return ujsonFrom(mapper.valueToTree(docs.get(0)));
+            }
+            return ujsonFrom(mapper.valueToTree(docs));
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new PluginException("Failed to read yaml data");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> Document<T> write(Value input, MediaType mediaType, Class<T> targetType) throws PluginException {
+        Charset charset = mediaType.getCharset();
+        if (charset == null) {
+            charset = Charset.defaultCharset();
+        }
+
+        try {
+            Object inputAsJava = ujsonUtils.javaObjectFrom(input);
+            ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+            StringBuilder value = null;
+
+            //if instance of list, it is multiple docs in one.
+            if(inputAsJava instanceof List){
```

Comment:
We'll need a writer parameter to tell us what to do here. Something like arrayAsMultiDoc

Reply:
What would we do instead in this case?
im assuming if `arrayAsMultiDoc` is false it would do something else, but what else could it do while still being valid yaml?

Reply:
It would come out as a single doc that's an array in YAML, those are legal top-levels, just like in JSON (some JSON processors restrict that, but the spec definitely allows it)